### ISSUE TITLE
integrate_personal_message has little useful effect

### ISSUE THREAD

**Issue by Archer70** [elkarte/Elkarte#2697]
I'm attempting to put together a mod that encrypts personal messages in the database, so obviously the first thing one does in that case is scope out the integration hooks available.

In this case, `integrate_personal_message` is somewhat ineffective due to it being called on the `$subject` and `$message` variables slightly too late. If I may direct your attention to [PersonalMessage.subs.php](https://github.com/elkarte/Elkarte/blob/development/sources/subs/PersonalMessage.subs.php#L534-L546), lines 534 through 546, you'll notice that variables `$htmlsubject` and `$htmlmessage` are the ones actually being saved in the database.

However, after the PM is saved, a notification email is sent using `$subject` and `$message` again.

Basically I'm proposing the integration hook be added above line 534, and we should be good to go.

**Comment by emanuele45** [comment]
Judging by the *hidden history* (i.e. commits from SMF pre-2.0) the original idea was just to let use PMs in other applications (something like the integrate_create_topic).
Then, at some point (it's interesting, there are pieces of code that change without any notice, I wonder why) the code was changed to let mods/addons modify the content of the variables, but the hook remained in a basically useless place.

I wouldn't call it a bug, but I feel it makes sense to move it up a bit and make it more useful.

**Comment by Archer70** [comment]
Great, thanks. I'll submit a PR later this afternoon, time permitting.
